modules/react/collection/stories/Collection.stories.mdx

```diff
@@ -53,13 +55,42 @@ item for selection, maintaining a cursor, or anything else.
 ### Dynamic Items
 
 The `ListBox` also handles a dynamic collection of items. Instead of providing each `ListBox.Item`
-statically, provide a render function instead. The function is called with an `items` value that is
-the same was what's provided to the model. By default, providing items will enable virtualization.
-This example adds a `maxHeight` to ensure overflow. Virtualization uses absolute positioning of each
-item, which could cause problems for popup menus. If your item count is low, pass
-`shouldVirtualize={false}` to disable virtualization.
+statically, provide a render function instead. The function is called with an `item` value that is
+the same as what's provided to the model.
 
-<ExampleCodeBlock code={DynamicItems} />
+#### Registering Items
+
+Dynamic items can be passed in as an array of strings or objects via the `items` prop or the model
+config.
+
+##### Array of Strings
+
+Use strings when your data is a simple list of options. Each string value is used as the unique
+identifer.
+
+<ExampleCodeBlock code={DynamicItemsStrings} />
+
+##### Array of Objects
+
+Use objects when your data contains additional information. Each object must have an `id` to ensure
+it's registered and searchable. Additional information such as icons or if the item is disabled can
+be included in the object and will be accessible in the render function.
+
+<ExampleCodeBlock code={DynamicItemsObjects} />
+
+> **Note: The `id` should match your text value for each item. This ensures proper selection of
+> items in components like `Select` and `Autocomplete`. If your `id` is different from your text
+> value, you'll have to add a
+> [translation layer](https://codesandbox.io/s/combobox-object-items-forked-mc7tm4). **
```

Comment:
I'd suggest removing `Autocomplete` since that implies we have an `Autocomplete` component in canvas-kit. The linked Codesandbox also has a few issues:

1. "Submist" should be "Submit".
2. The item `label`s should be fruits instead of Foo, Bar, etc.
3. I'm hoping this is a bug with the example and not the component, but I noticed the up/down arrow keys don't work for navigating the menu.

Reply:
3. ... not sure how to fix that. The keyboard handler is using `id` and not the label

Reply:
@NicholasBoll I'm not sure this will ever work since keyboard handling depends on the id

Reply:
The example was missing the `getId` function. It is fixed now.

Reply:
This also raises the issue that `getId` and `getTextValue` are never mentioned in the documentation.

Reply:
i can add that